### PR TITLE
Remove trademarked language in code comments

### LINKED ISSUE #148: Occurrences of "Redis" in comments
* Don't touch the comments where Redis is mentioned in Copyright and License text. ("Neither the name of Redis nor the names of its contributors may be used ...")
* Where "Redis Cluster" or "Redis Sentinel" is mentioned, we can just replace it with "Cluster" or "Sentinel"
* In some cases, "Redis" can be replaced with a generic word like "the server" or just skipped, depending on context. Examples:
  * "the Redis cluster-bus" => "the cluster-bus"
  * "the Redis command" => "the command"
  * "the currently SELECTed Redis DB" => "the currently SELECTed DB"
  * "the Redis ..." => "the ..." (most of the time)
* In most other cases, just replace "Redis" with "Valkey"
* Don't change code under deps/

### PR DESCRIPTION
**Pull request by murphyjacob4**
Strategy for replacement:
* Regex search: `(//|/\*| \*|#).* ("|\()?(r|R)edis( |\. |'|\n|,|-|\)|")(?!nor the names of its contributors)(?!Ltd.)(?!Labs)(?!Contributors.)`
* Don't edit copyright comments
* Replace "Redis version X.X" -> "Redis OSS version X.X" to distinguish from newly licensed repository
* Replace "Redis Object" -> "Object"
* Exclude markdown for now
* Don''t edit Lua scripting comments referring to redis.X API
* Replace "Redis Protocol" -> "RESP"
* Replace redis-benchmark, -cli, -server, -check-aof/rdb with "valkey-" prefix
* Most other places, I use best judgement to either remove "Redis", or replace with "the server" or "server"

Fixes #148

### CODE REVIEW

File: deps/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Redis dependency Makefile
+# Server dependency Makefile
```
Comment (zuiderkwast) [inline deps/Makefile]: Actually some dependencies are for valkey-cli and valkey-benchmark. Specifically hiredis is a client lib.

```suggestion
# Dependency Makefile
```

File: deps/hiredis/.github/workflows/test.yml [hunk 2]
```diff
@@ -92,8 +92,8 @@ jobs:
         run: MSBuild hiredis.vcxproj /p:Configuration=Debug
       - name: Build hiredis-test
         run: MSBuild hiredis-test.vcxproj /p:Configuration=Debug
-      # use memurai, redis compatible server, since it is easy to install.  Can't
-      # install official redis containers on the windows runner
+      # use memurai, a Redis protocol compatible server, since it is easy to install.  Can't
+      # install official containers on the windows runner
```
Comment (zuiderkwast) [inline deps/hiredis/.github/workflows/test.yml]: Please don't change files under deps/hiredis/. It's a vendored dependency and we may want to be able to update it in the future.

I actually mentioned it in the issue "Don't change code under deps/".

File: deps/jemalloc/include/jemalloc/jemalloc_macros.h.in [hunk 3]
```diff
@@ -148,6 +148,6 @@
 #  define JEMALLOC_SYS_NOTHROW JEMALLOC_NOTHROW
 #endif
 
-/* This version of Jemalloc, modified for Redis, has the je_get_defrag_hint()
+/* This version of Jemalloc, modified for this project, has the je_get_defrag_hint()
```
Comment (zuiderkwast) [inline deps/jemalloc/include/jemalloc/jemalloc_macros.h.in]: It's a little ambigous what "this project" means with this vendored dependency. If it's part of Jemalloc, "the project" could mean the Jemalloc project.

To avoid the confusion, we can either keep Redis (since it was actually modified for Redis) or change it to Valkey. I'm leaning towards leaving it as Redis in this case actually.

File: src/acl.c [hunk 4]
```diff
@@ -3204,7 +3204,7 @@ void addReplyCommandCategories(client *c, struct serverCommand *cmd) {
 }
 
 /* AUTH <password>
- * AUTH <username> <password> (Redis >= 6.0 form)
+ * AUTH <username> <password> (Redis OSS >= 6.0 form)
```
Comment (zuiderkwast) [inline src/acl.c:3207]: Just marking this for discussion.

Do you actually believe forked code like this can be a trademark issue?

I don't mind that we're changing it to Redis OSS though we usually want to avoid unnecessary changes. Let's leave this open for comments. I want to know what others think.

File: src/db.c [hunk 5]
```diff
@@ -885,7 +885,7 @@ void scanCallback(void *privdata, const dictEntry *de) {
     serverAssert(!((data->type != LLONG_MAX) && o));
 
     /* Filter an element if it isn't the type we want. */
-    /* TODO: uncomment in redis 8.0
+    /* TODO: uncomment in version 8.0
```
Comment (zuiderkwast) [inline src/db.c:888]: Actually we should uncomment these in version 8.0

Let's open an issue for this. There are more occurrences of "TODO: uncomment in version 8.0" in this file (`db.c`).

For more information, see the commit 07ed0eafa98a66796248b3759aa3cf3d0eb138ec `improve performance for scan command when matching pattern or data type (#12209)`

File: src/dict.c [hunk 6]
```diff
@@ -50,7 +50,7 @@
 
 /* Using dictSetResizeEnabled() we make possible to disable
  * resizing and rehashing of the hash table as needed. This is very important
- * for Redis, as we use copy-on-write and don't want to move too much memory
+ * for the server, as we use copy-on-write and don't want to move too much memory
```
Comment (zuiderkwast) [inline src/dict.c]: Minor, just an idea

```suggestion
 * for us, as we use copy-on-write and don't want to move too much memory
```

File: src/eval.c [hunk 7]
```diff
@@ -1508,9 +1508,9 @@ void ldbEval(lua_State *lua, sds *argv, int argc) {
     lua_pop(lua,1);
 }
 
-/* Implement the debugger "redis" command. We use a trick in order to make
+/* Implement the debugger "server command". We use a trick in order to make
```
Comment (zuiderkwast) [inline src/eval.c]: I've never used the Lua debugger in Redis so I don't really know what this is supposed to mean, but reading the source code below, I believe it calls the Lua function `redis.call`. We are adding a synonym for that called `server.call` but the debugger implementation below still uses redis.call and honestly i don't think it matters. Anyway, this PR is only about the comments so... perhaps it's just the quotation mark that's a bit off:

```suggestion
/* Implement the debugger "server" command. We use a trick in order to make
```

File: src/module.c [hunk 8]
```diff
@@ -1583,15 +1583,15 @@ int RM_SetCommandACLCategories(RedisModuleCommand *command, const char *aclflags
  *
  * All fields except `version` are optional. Explanation of the fields:
  *
- * - `version`: This field enables compatibility with different Redis versions.
+ * - `version`: This field enables compatibility with different server versions.
  *   Always set this field to REDISMODULE_COMMAND_INFO_VERSION.
```
Comment (zuiderkwast) [inline src/module.c]: This is a tricky one. We'll have to track both redis version compatibility and valkey version compatibility and we're talking about introducing a ValkeyModule API... TBD

**Review by zuiderkwast — CHANGES_REQUESTED** [review]
Partial review. I made it to the middle of `module.c` so far.

File: src/acl.c (hunk 4 above)
Comment (murphyjacob4) [reply src/acl.c:3207]: IANAL - but it probably wouldn't be an issue to leave it.

However - with the license changes, Redis was clear on what naming they wanted moving forward for "Redis Community Edition" vs "Redis OSS": https://redis.com/blog/redis-adopts-dual-source-available-licensing/

> We have referred to versions of our products as either open source (OSS), Enterprise, or Cloud. Redis v7.2 and prior releases will continue to be called Redis OSS. Starting with Redis v7.4, we will refer to the open, freely available versions as Redis Community Edition. Both the RSALv2 and SSPL v1 licenses provide open and free access to the underlying source code of Redis Community Edition. However, it does not meet the definition of open source as defined by OSI, and that is why we will refer to the products as the “community edition” rather than the “open source” edition, as we did previously. There are many references to open source on our websites, and we are actively working to clarify these language changes in the coming weeks.

To me - it makes sense to align our nomenclature. But yeah, happy to keep this open for discussion.

File: deps/jemalloc/include/jemalloc/jemalloc_macros.h.in (hunk 3 above)
Comment (murphyjacob4) [reply deps/jemalloc/include/jemalloc/jemalloc_macros.h.in]: Makes sense. Reverted the change

File: deps/hiredis/.github/workflows/test.yml (hunk 2 above)
Comment (murphyjacob4) [reply deps/hiredis/.github/workflows/test.yml]: Apologies, I missed adding that exclusion to my search. Reverted 👍

File: deps/Makefile (hunk 1 above)
Comment (murphyjacob4) [reply deps/Makefile]: Applied

File: src/module.c (hunk 8 above)
Comment (murphyjacob4) [reply src/module.c]: So what are you thinking - should we leave the comment as is - since we need a bigger decision about what this field means? Sounds like it is possible we will leave this "version" to be used for "Redis version" (i.e. locked at 7.2.4)

File: src/db.c (hunk 5 above)
Comment (murphyjacob4) [reply src/db.c:888]: Filed #235

File: src/module.c (hunk 8 above)
Comment (zuiderkwast) [reply src/module.c]: Actually no problem. This macro only defines the version of the struct defined in the same header file.

File: src/module.c [hunk 9]
```diff
@@ -4522,8 +4522,7 @@ int moduleListIteratorSeek(RedisModuleKey *key, long index, int mode) {
  * - EINVAL if key or ele is NULL.
  * - ENOTSUP if the key is of another type than list.
  * - EBADF if the key is not opened for writing.
- *
- * Note: Before Redis 7.0, `errno` was not set by this function. */
+ */
```
Comment (zuiderkwast) [inline src/module.c]: Please keep this note and use "Redis OSS".

These comments in `module.c` are used for generated documentation of the module API, so in this case I think we should use the correct trademark term "Redis OSS". These notes about old versions are useful for module authors.

FYI: Here is the markdown file generated from the comments: https://github.com/valkey-io/valkey-doc/blob/main/docs/reference/modules/modules-api-ref.md. A script under utils/ is used for that.

File: src/module.c [hunk 10]
```diff
@@ -7672,13 +7671,13 @@ void RM_LogIOError(RedisModuleIO *io, const char *levelstr, const char *fmt, ...
     va_end(ap);
 }
 
-/* Redis-like assert function.
+/* serverAssert-like function.
```
Comment (zuiderkwast) [inline src/module.c]: `redisAssert` is an internal function. It's not familiar to module API users. Better something like this...

```suggestion
/* Valkey assert function.
```

File: src/networking.c [hunk 11]
```diff
@@ -476,7 +476,7 @@ void addReplyProto(client *c, const char *s, size_t len) {
 }
 
 /* Low level function called by the addReplyError...() functions.
- * It emits the protocol for a Redis error, in the form:
+ * It emits the protocol for a server error, in the form:
```
Comment (zuiderkwast) [inline src/networking.c]:
```suggestion
 * It emits the protocol for an error reply, in the form:
```

This can be a client error or a server error.

A server error usually means that something failed on the server side. For example, the HTTP error codes use 400 for client errors, like trying to access a file not found (404), while server errors are 500 series (e.g. 500 internal server error).

File: src/sentinel.c [hunk 12]
```diff
@@ -617,7 +617,7 @@ int sentinelAddrEqualsHostname(sentinelAddr *a, char *hostname) {
                     sentinel.resolve_hostnames ? ANET_NONE : ANET_IP_ONLY) == ANET_ERR) {
 
         /* If failed resolve then compare based on hostnames. That is our best effort as
-         * long as the server is unavailable for some reason. It is fine since Redis 
+         * long as the server is unavailable for some reason. It is fine since a
```
Comment (zuiderkwast) [inline src/sentinel.c]: an instance

```suggestion
         * long as the server is unavailable for some reason. It is fine since an
```

File: src/server.h [hunk 13]
```diff
@@ -699,17 +699,17 @@ typedef enum {
  * Data types
  *----------------------------------------------------------------------------*/
 
-/* A redis object, that is a type able to hold a string / list / set */
+/* An Object, that is a type able to hold a string / list / set */
 
-/* The actual Redis Object */
+/* The actua Object */
```
Comment (zuiderkwast) [inline src/server.h]:
```suggestion
/* The actual Object */
```

File: tests/integration/logging.tcl [hunk 14]
```diff
@@ -8,7 +8,7 @@ if {$system_name eq {linux}} {
 }
 
 # look for the DEBUG command in the backtrace, used when we triggered
-# a stack trace print while we know redis is running that command.
+# a stack trace print while we know the server4 is running that command.
```
Comment (zuiderkwast) [inline tests/integration/logging.tcl]:
```suggestion
# a stack trace print while we know the server is running that command.
```

File: tests/unit/scan.tcl [hunk 15]
```diff
@@ -109,7 +109,7 @@ proc test_scan {type} {
 
         after 2
 
-        # TODO: remove this in redis 8.0
+        # TODO: remove this in server version 8.0
```
Comment (zuiderkwast) [inline tests/unit/scan.tcl:112]: Are these TODOs covered by that issue you created for some other TODOs? I didn't check. There are a few of these in this test suite.

**Review by zuiderkwast — COMMENTED** [review]
I went through all of it now. Thanks for this massive work!

I'm sorry you got merge conflicts now, probably a lot of them. I hope they're easy to solve though.

File: src/cluster_legacy.c [hunk 16]
```diff
@@ -169,7 +169,7 @@ dictType clusterSdsToListType = {
         NULL                        /* allow to expand */
 };
 
-/* Aux fields are introduced in Redis 7.2 to support the persistence
+/* Aux fields were introduced to support the persistence
```
Comment (zuiderkwast) [inline src/cluster_legacy.c]:
```suggestion
/* Aux fields were introduced in Redis OSS 7.2 to support the persistence
```

I think it can be good to keep this, since it's fairly new. In a cluster, nodes need to communicate with nodes running different versions.

File: src/module.c (hunk 9 above)
Comment (murphyjacob4) [reply src/module.c]: Gotcha. Yeah on my first pass I removed these notes, then later realized they needed to be kept, and tried to add them back. Must have missed this one.

File: tests/unit/scan.tcl (hunk 15 above)
Comment (murphyjacob4) [reply tests/unit/scan.tcl:112]: Yeah I believe these are tests related to the proposed SCAN changes

**Comment by murphyjacob4** [comment]
Thanks for the thorough review! Merge conflicts weren't too bad

**Comment by zuiderkwast** [comment]
Please check the failed CI jobs. It seems some merge conflicts were not solved in the right way...

**Comment by murphyjacob4** [comment]
Yeah building is important... :)

Had a comment that got double terminated after the merge. `make` and `make test` should be fixed now

**Comment by murphyjacob4** [comment]
Resolved some more merge conflicts

File: src/acl.c (hunk 4 above)
Comment (murphyjacob4) [reply src/acl.c:3207]: @zuiderkwast - anyone in particular you want to involve in the discussion?

Comment (zuiderkwast) [reply src/acl.c:3207]: Not really. You're right. Let's use Redis OSS. :+1:

File: src/module.c [hunk 17]
```diff
@@ -1249,7 +1249,7 @@ ValkeyModuleCommand *moduleCreateCommandProxy(struct ValkeyModule *module, sds d
  *                          the arguments that are channels.
  *
  * The last three parameters specify which arguments of the new command are
- * Redis keys. See https://redis.io/commands/command for more information.
+ * keys. See https://redis.io/commands/command for more information.
```
Comment (zuiderkwast) [inline src/module.c]: Let's just change this too. I hope the URL will work in our new website.

```suggestion
 * keys. See https://valkey.io/commands/command for more information.
```

**Review by zuiderkwast — APPROVED** [review]
Really awesome. I'll try to merge this fast before any other changes to avoid more merge conflict.
